src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final float GROW_FACTOR = 1.5F;
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] values;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= values.length) {
+            values = grow();
+        }
+        values[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == 0 && index == size) {
+            add(value);
+        } else {
```

Comment:
let's call return after add - this way we won't need to wrap the rest of the logic in `else` - will be more readable, it's a [known pattern](https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement)